src/time_steppers.jl

```diff
@@ -104,6 +96,22 @@ end
 
 time_step!(model; Nt, Δt) = time_step!(model, Nt, Δt)
 
+function solve_for_pressure!(::CPU, model::Model)
+    Nx, Ny, Nz = model.grid.Nx, model.grid.Ny, model.grid.Nz
+    RHS, ϕ = model.poisson_solver.storage, model.poisson_solver.storage
+
+    solve_poisson_3d_ppn_planned!(model.poisson_solver, model.grid)
+    @views model.pressures.pNHS.data[1:Nx, 1:Ny, 1:Nz] .= real.(ϕ)
```

Comment:
view view view

Reply:
note that a view is created each time-step here. We could pre-allocate the view and re-use it on this line. Perhaps an enhancement for the future.